driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -132,21 +118,23 @@ public synchronized void reset()
     {
         ensureSessionIsOpen();
         ensureNoUnrecoverableError();
-        ensureConnectionIsOpen();
 
         if ( currentTransaction != null )
         {
             currentTransaction.markToClose();
             lastBookmark = currentTransaction.bookmark();
             currentTransaction = null;
         }
-        connection.resetAsync();
+        if ( currentConnection != null )
+        {
+            currentConnection.resetAsync();
```

Comment:
I would suggest to remove reset. Make this method deprecated and then give an empty impl inside.